### PR TITLE
[mds-agency] [mds-db] Initial Stop Prototype

### PR DESCRIPTION
**Pull request by avatarneil**
Initial prototype implementation of Stops in MDS, as a superset of both GTFS stops and GBFS docks. 
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [x] write tests for all new functionality

## Impacts
- [ ] Provider
- [x] Agency
- [ ] Audit
- [ ] Policy
- [ ] Compliance
- [ ] Daily
- [ ] Native
- [ ] Policy Author

### CODE REVIEW

File: packages/mds-agency/request-handlers.ts [hunk 1]
```diff
@@ -447,3 +448,49 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
...
   }
 }
+
+export const registerStop = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+  const stop = req.body
+  const { geography_id } = stop
+
+  try {
+    isValidStop(stop)
+    await readSingleGeography(geography_id)
```
Comment (janedotx) [inline packages/mds-agency/request-handlers.ts]: Why is this in here? You don't use the result at all.

Comment (janedotx) [reply packages/mds-agency/request-handlers.ts]: Never mind, I see, it throws if it doesn't find it.

File: packages/mds-db/stops.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    rows: [recorded_stop]
+  }: { rows: Recorded<Stop>[] } = await client.query(sql, values)
+  return { ...stop, ...recorded_stop }
+}
+
+export async function readStop(stop_id: UUID): Promise<Recorded<Stop>> {
+  const client = await getReadOnlyClient()
+  const sql = `SELECT * FROM ${schema.TABLE.stops} WHERE ${schema.COLUMN.stop_id} = '${stop_id}'`
+  const {
+    rows: [recorded_stop]
```
Comment (janedotx) [inline packages/mds-db/stops.ts]: Ok what TS/JS language feature is this? I know what the square brackets do when they're describing the key, but I'm not sure what they mean when it's the value like this.

**Comment by janedotx** [comment]
Looks like a great draft so far.

File: packages/mds-db/stops.ts (hunk 2 above)
Comment (avatarneil) [reply packages/mds-db/stops.ts]: https://www.typescriptlang.org/docs/handbook/variable-declarations.html#destructuring

Comment (avatarneil) [reply packages/mds-db/stops.ts]: Basically, we're destructuring the `rows` array from the SQL response, and pulling the first returned member into the `recorded_stop` variable.

File: packages/mds-agency/request-handlers.ts [hunk 3]
```diff
@@ -447,3 +448,47 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
...
+  try {
+    const recorded_stop = await db.readStop(stop_id)
+
+    if (!recorded_stop) {
+      throw new NotFoundError()
+    }
+
+    res.status(200).send(recorded_stop)
+  } catch (err) {
+    res.status(404).send(err)
```
Comment (BigIdeas) [inline packages/mds-agency/request-handlers.ts]: Are you sure `db.readStop` won't throw an error trying to connect to or read from the db (in which case we'd want to return 500 here)?

File: packages/mds-agency/request-handlers.ts [hunk 4]
```diff
@@ -447,3 +448,47 @@ export const submitVehicleTelemetry = async (req: AgencyApiRequest, res: AgencyA
...
+  }
+}
+
+export const readStops = async (req: AgencyApiRequest, res: AgencyApiResponse) => {
+  const stops = await db.readStops()
```
Comment (BigIdeas) [inline packages/mds-agency/request-handlers.ts]: Similar comment - could `db.readStops` throw an exception?

File: packages/mds-agency/request-handlers.ts (hunk 3 above)
Comment (drtyh2o) [reply packages/mds-agency/request-handlers.ts]:
```
if (!recorded_stop) {
  return res.status(404).send(new NotFoundError())
}
```
Then you can return 500 in the catch.

File: packages/mds-db/schema.ts [hunk 5]
```diff
@@ -14,14 +14,16 @@ const TABLE = Enum(
   'migrations',
   'policies',
   'policy_metadata',
-  'telemetry'
+  'telemetry',
+  'stops'
```
Comment (drtyh2o) [inline packages/mds-db/schema.ts]: This list looks like it was intended to be sorted.

File: packages/mds-db/schema.ts [hunk 6]
```diff
@@ -173,6 +193,27 @@ const TABLE_COLUMNS: { [T in TABLE_NAME]: Readonly<COLUMN_NAME[]> } = {
     COLUMN.altitude,
     COLUMN.charge,
     COLUMN.recorded
+  ],
+  [TABLE.stops]: [
+    COLUMN.stop_id,
```
Comment (drtyh2o) [inline packages/mds-db/schema.ts]: Like other tables, this should have an `id` (IDENTITY) column.

File: packages/mds-db/stops.ts [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  await logSql(sql, values)
+  const {
+    rows: [recorded_stop]
+  }: { rows: Recorded<Stop>[] } = await client.query(sql, values)
+  return { ...stop, ...recorded_stop }
+}
+
+export async function readStop(stop_id: UUID): Promise<Recorded<Stop>> {
+  const client = await getReadOnlyClient()
+  const sql = `SELECT * FROM ${schema.TABLE.stops} WHERE ${schema.COLUMN.stop_id} = '${stop_id}'`
```
Comment (drtyh2o) [inline packages/mds-db/stops.ts]: Should use `SqlVals` instead of string templates.

File: packages/mds-agency/tests/db-integration-tests.ts [hunk 8]
```diff
@@ -57,7 +59,7 @@ function now(): Timestamp {
 
 const APP_JSON = 'application/json; charset=utf-8'
 
-const LA_CITY_BOUNDARY = '1f943d59-ccc9-4d91-b6e2-0c5e771cbc49'
+const LA_CITY_BOUNDARY_ID = '1f943d59-ccc9-4d91-b6e2-0c5e771cbc49'
```
Comment (drtyh2o) [inline packages/mds-agency/tests/db-integration-tests.ts]: Seems like for testing purposes, these IDs and such could be city agnostic.

File: packages/mds-schema-validators/validators.ts [hunk 9]
```diff
@@ -209,6 +210,42 @@ const auditIssueCodeSchema = stringSchema.max(31)
 
 const auditNoteSchema = stringSchema.max(255)
 
+const vehicleTypesCountMapSchema = Joi.object().keys({
+  scooter: Joi.number(),
+  bike: Joi.number(),
+  carshare: Joi.number()
```
Comment (drtyh2o) [inline packages/mds-schema-validators/validators.ts]: Should this include all vehicle types or no?

Comment (avatarneil) [reply packages/mds-schema-validators/validators.ts]: It should indeed -- with the switch of `carshare -> car` and addition of `moped` this needs to be updated

File: packages/mds-agency/tests/db-integration-tests.ts (hunk 8 above)
Comment (avatarneil) [reply packages/mds-agency/tests/db-integration-tests.ts]: This is the same UUID we use for the greater CoLA service area, so it's not really city agnostic at this point. Once we fully deprecate service areas and come up with some generic geospatial fixtures, this should be city agnostic.

File: packages/mds-db/schema.ts [hunk 10]
```diff
@@ -173,6 +193,28 @@ const TABLE_COLUMNS: { [T in TABLE_NAME]: Readonly<COLUMN_NAME[]> } = {
...
+    COLUMN.capacity,
+    COLUMN.location_type,
+    COLUMN.timezone,
+    COLUMN.cross_street,
+    COLUMN.num_vehicles_available,
+    COLUMN.num_vehicles_disabled,
+    COLUMN.num_spots_available,
+    COLUMN.num_spots_disabled,
+    COLUMN.wheelchair_boarding,
+    COLUMN.reservation_cost
```
Comment (BigIdeas) [inline packages/mds-db/schema.ts]: Do we want a `COLUMN.recorded` on this table?

**Pull request merged**